src/find-menu.js

```diff
@@ -26,42 +33,171 @@ class FindMenu extends HTMLElement {
     await this.loadSVG(this.nextButton, 'peersky://static/assets/svg/down.svg');
     await this.loadSVG(this.hideButton, 'peersky://static/assets/svg/close.svg');
 
+    // Setup foundInPage listener on webview
+    this.setupFoundInPageListener();
+    
+    // Setup webview navigation events to detect PDFs
+    this.setupWebviewNavigationListener();
+
     this.input.addEventListener('input', (e) => {
       const { value } = this;
-      if (!value) return;
-      if (value.length > 0 && value !== this.currentSearchValue) {
-        this.currentSearchValue = value;
-        this.resetSearch();
+      if (!value) {
+        this.stopFindInPage('clearSelection');
+        return;
       }
-      this.dispatchEvent(new CustomEvent('next', { detail: { value } }));
+      
+      this.findInWebview(value, { forward: true });
```

Comment:
Currently, the search seems to only update when Enter is pressed. To improve user experience, we should update results dynamically as the user types by setting `findNext: true` in `findInWebview()`